mars/oscar/core.pyx

```diff
@@ -178,6 +210,104 @@ cdef class ActorRefMethod:
         return asyncio.create_task(delay_fun())
 
 
+cdef class ActorLocalRef(ActorRef):
+    def __init__(self, _BaseActor actor):
+        # Make sure the input actor is an instance of _BaseActor.
+        super().__init__(actor._address, actor._uid)
+        self._actor_weakref = weakref.ref(actor, lambda _: self._methods.clear())
+
+    cdef _weakref_local_actor(self):
+        actor = _get_local_actor(self.address, self.uid)
+        # Make sure the input actor is an instance of _BaseActor.
+        if actor is not None and isinstance(actor, _BaseActor):
+            self._actor_weakref = weakref.ref(actor, lambda _: self._methods.clear())
+            return actor
+        return None
+
+    def __getattr__(self, item):
+        try:
+            return self._methods[item]
+        except KeyError:
+            actor = self._actor_weakref() or self._weakref_local_actor()
+            if actor is None:
+                raise ActorNotExist(f"Actor {self.uid} does not exist") from None
+            getattr(actor, item)
```

Comment:
This line seems useless?

Reply:
It detects the attribute error as early as possible. While, the `ActorRef` object does not has the actor metadata, so it can't check the attribute error when getting the attribute.

Reply:
Then add comments about it.